scalajvm/app/helpers/ReCaptcha.scala

```diff
@@ -1,24 +1,25 @@
 package helpers
 
-import com.netaporter.uri.dsl._
-import dispatch._, Defaults._
-import play.api.Play.current
 import play.libs.Json
+import play.api.Configuration
+import javax.inject._
+import scalaj.http._
 
-import scala.concurrent.Future
+import scala.concurrent.{Future, blocking}
+import scala.concurrent.ExecutionContext.Implicits.global
 
-object ReCaptcha {
+@Singleton
+class ReCaptcha @Inject()(implicit configuration: Configuration) {
 
-  lazy val publicKey = current.configuration.getString("recaptcha.publickey").get
-  lazy val privateKey = current.configuration.getString("recaptcha.privatekey").get
+  lazy val publicKey = configuration.get[String]("recaptcha.publickey")
+  lazy val privateKey = configuration.get[String]("recaptcha.privatekey")
 
   val verificationUrl = "https://www.google.com/recaptcha/api/siteverify"
 
   def check(addr: String, response: String): Future[Boolean] = {
-    val uri = verificationUrl ? ("secret" -> privateKey) & ("response" -> response) & ("remoteip" -> addr)
-    val request = url(uri)
-    val result = Http(request OK as.String)
+    val request = Http(verificationUrl).params("secret" -> privateKey, "response" -> response, "remoteip" -> addr)
+    val result = Future { blocking { request.asString.body } }
```

Comment:
I don't get the purpose of `blocking` here. Could you please explain? It compiles without `blocking`, and I doubt we need it here.

Reply:
scalaj-http that we use is a synchronous library, so all calls to it are blocking. From [Scala docs on futures](https://docs.scala-lang.org/overviews/core/futures.html#blocking) I gather that futures are expected to use non-blocking operations at their lowest level. Since the future I'm constructing here is not like that, I thought it fitting to mark the code accordingly.

The doc mentions that this call might have no effect. I don't really know when it would, but I think it makes sense to keep it in case we move to a different executor that e.g. spawns individual threads for each blocking call.

Reply:
Actually, I should probably put that explanation into a comment. Let me do that.

Reply:
Alright, this is good.